proposals/07-datastore-optimization.md

```diff
@@ -0,0 +1,101 @@
+
+The aim here is to finalize existing explorations to deliver at least a 50% improvement to chain performance.
+
+#### Assumptions &amp; hypotheses
+
+* Datastore performance affects chain sync speed (we have experimental evidence this is true, but not indications of how much)
+* Prototyped solutions that demonstrate improvement as prototypes will not get slowed down a lot in getting to production readiness
+* Chain performance raises the barrier to entry and creates negative characteristics that cause people to leave Filecoin (see:  high gas fees, long time to get up and running without snapshot)
+* A 50% improvement that isn’t an order of magnitude improvement is still worth pursuing, especially if we have a clear path to success
+* Existing work that hasn’t been finished and benchmarked will deliver at least modest improvements
+
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+#### Impact
+
+🔥🔥
+
+Chain speed breaks everything. Even 50% improvement has a big impact on everyone using filecoin.
+
+#### Leverage
+
+🎯🎯
+
+We are specifically picking this as an easy win on the datastore improvement track. However, it might be something we bring to IPFS eventually, and it will set us up to build more specialized IPLD solutions.
+
+#### Confidence
+
+3 - All user reports that the hardware requirements for running a full lotus node are a major pain point -- if we could drop those requirements, even a little, it would make a big difference
+
+## Project definition
+#### Brief plan of attack
+
+We have three significant potential solutions that are well prototyped and close to delivery --
+ * [Using a custom append only key value store](https://github.com/filecoin-project/lotus/pull/5575)
+ * [Split store](https://github.com/filecoin-project/lotus/pull/4992)
+ * [Archival store](https://github.com/ribasushi/ltsh/tree/psql_fullcapture)
+
+The plan of attack is to aggressively test these solutions to determine the best combination of data store, including potentially a segregated choice between hot and cold storage, and deliver a solution that can become the default for Lotus.
+
+* Lotus CLI extended to allow a command line flag selection of datastore
+* Support code for split store, custom key-value, and archival chain store as in-repo options so they are maintained
+* Tests of functionality for all backends to mitigate future breakage / code rot.
+
+#### What does done look like?
+
+A released version of lotus with faster chainsync
+
+####  What does success look like?
+
+Success means drop in gas fees as more messages fit in a block
+
+Success means being able to slightly lower system requirements to run a lotus full node
```

Comment:
Would this also include being able to sync the chain faster, or just a sync’d chain taking up less disk space?

Reply:
faster